app/tests/test_routes.py

```diff
@@ -0,0 +1,73 @@
+def test_get_all_returns_200_and_empty_array(client):
+    response = client.get("/planets")
+    response_body = response.get_json()
+
+    assert response.status_code == 200
+    assert response_body == []
+
+
+def test_get_one_planet(client, sample_data_with_two_planets):
+    # Act
+    response = client.get("/planets/1")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
+    assert response_body == {"id":1,
+                            "name":"Ocean Planet",
+                            "description":"It's wet!",
+                            "has_moons":True}
+
+def test_get_one_planet_without_data_returns_404(client):
+    # Act
+    response = client.get("/planets/1")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 404
+    assert response_body == {"details":"Planet id: 1 not found"}
+
+def test_get_all_planets(client, sample_data_with_two_planets):
+    # Act
+    response = client.get("/planets")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
+    assert response_body == [{"id":1,
+                            "name":"Ocean Planet",
+                            "description":"It's wet!",
+                            "has_moons":True},
+                            {"id":2,
+                            "name":"Jungle Planet",
+                            "description":"Full of big cats. And trees!",
+                            "has_moons":False}] 
+
+def test_create_a_planet(client):
+    # Act
+    response = client.post("/planets", json={
+        "name":"Ice Planet",
+        "description":"Right next to Canada",
+        "has_moons":True
+    })
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 201
+    assert response_body == {"details":"Planet Ice Planet successfully added to the Planets Database."}
+
+def test_update_planet_patches_properly(client, sample_data_with_two_planets):
+    pre_patch_response = client.get("/planets/1")
```

Comment:
We generally want to start with hardcoded data rather than the result of a call to our application. While ideally there wouldn't be, if there were a bug in the GET route it could affect this test's ability to accurately test what we want/hide issues with the route we're actively testing.